doc/nrf/app_dev/device_guides/nrf54l/building_nrf54l.rst

```diff
@@ -0,0 +1,120 @@
+.. _building_nrf54l:
+
+Building and programming with nRF54L15 DK
+#########################################
+
+.. contents::
+   :local:
+   :depth: 2
+
+This guide provides instructions on how to build and program the nRF54L15 development kit.
+Whether you are working with single or multi-image builds, the following sections will guide you through the necessary steps.
+
+Depending on the sample, you must program only the application core or both the Fast Lightweight Peripheral Processor (FLPR) and the application core.
+Additionally, the process will differ based on whether you are working with a single-image or multi-image build.
+
+This process complements the standard building procedure.
+While building for the application core follows the default process, if you wish to utilize the FLPR core you must use additional configuration to enable it.
+
+Building for the application core
+*********************************
+
+For instructions on building for the application core only, see how to :ref:`build an application <building>`.
+
+Building for the application and FLPR core
+******************************************
+
+This section outlines how to build and program for both the application and FLPR core, covering separate builds and sysbuild configurations.
+FLPR core supports two variants:
+
+* ``nrf54l15dk/nrf54l15/cpuflpr`` where FLPR runs from SRAM, which is the recommended method.
+  To build FLPR image with this variant, the application core image must include the ``nordic-flpr`` snippet.
+
+* ``nrf54l15dk/nrf54l15/cpuflpr/xip`` where FLPR runs from RRAM.
+  To build FLPR image with this variant, the application core image must include the ``nordic-flpr-xip`` snippet.
+
+Separate images
+---------------
+
+To build and program the application sample and the FLPR sample as separate images, follow these steps:
+
+.. tabs::
+
+   .. group-tab:: nRF Connect for VS Code
+
+      .. include:: /includes/vsc_build_and_run.txt
+
+      3. Set the following options:
+
+         * Board target to ``nrf54l15dk/nrf54l15/cpuflpr`` (recommended) or ``nrf54l15dk/nrf54l15/cpuflpr/xip``.
+
+         * Appropriate snippet depending on the selected board target.
```

Comment:
1. The recommended snippets selection does not align with the guide for building using CLI.
Appropriate snippets should be selected for the application's image, not for the flpr (or flpr/xip).
2. I miss another section explaining how to build the firmware for the application core.
3. Please, put a note that the VS Code extension does not fully support the FLPR on nRF54L15 - the users should not expect at least the debugging feature to work. I suggest to mark every section that includes any action concerning flpr and vscode as experimental (or not fully supported).

Reply:
Ad. 1. Those snippets should be selected for application's image, see:
`the application core image must include the [...] snippet`
There are 2 snippets because FLPR can run either from SRAM or RRAM, which causes some overhead in devicetree and different handling on APP core side (when FLPR is running from RRAM, application core can skip copying code to SRAM).

Reply:
I think the fact that this step is just under selecting board target to FLPR might be confusing.
It should be reiterated that the snippet is applied to app core image.

Reply:
I edited this section a bit to reflect this better + added a VS code support note - @FilipZajdel , @masz-nordic please double-check